src/wallet/wallet2.cpp

```diff
@@ -3025,6 +3035,8 @@ bool wallet2::load_keys(const std::string& keys_file_name, const epee::wipeable_
     m_account.set_device(hwdev);
     LOG_PRINT_L0("Device inited...");
   }
+  if (encrypted_secret_keys)
+    m_account.decrypt_keys(key);
```

Comment:
How about adding 
```c++
  else
    store_keys(keys_file_name, password, m_watch_only);
```
here to make sure this scheme is enabled when loading the old unencrypted format?

Reply:
Good idea.

Reply:
done

Reply:
and now done only when load success...